microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -223,6 +242,133 @@ def verify_kdumpcrash_large_memory_auto_size(
         self.is_auto = True
         self._kdump_test(node, log_path, log)
 
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. replace FRE bins with CHK bins and reboot the VM
+        2. Configure kdump and reboot VM
+        3. Generate Crash with hvdbg syscall and verify dump
+
+        The test expects the directory containing MSHV CHK binaries tar to be passed
+        in the mshv_chk_bin / mshv_chk_loader testcase variables.
+        """,
+        priority=2,
+    )
+    def verify_mshv_crash(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any],
+        log_path: Path,
+    ) -> None:
+        chkbinpath = variables.get("mshv_chk_bin", "")
+        chkloaderpath = variables.get("mshv_chk_loader", "")
+        if not chkbinpath or not chkloaderpath:
+            raise SkippedException(
+                "Requires a path to MSHV binaries to be passed. "
+                "Please set mshv_chk_bin and mshv_chk_loader testcase variable."
+            )
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                f"Testcase only support Mariner-MSHV. Found: {node.os}"
+            )
+
+        mshv = node.tools[Ls].path_exists("/dev/mshv", sudo=True)
+        if not mshv:
+            raise SkippedException("MSHV not present")
+
+        log.info(f"chkbinpath: {chkbinpath}")
+        log.info(f"chkloaderpath: {chkloaderpath}")
+
+        # Copy and Extract CHK tar on node
+        grub_config_file = "/boot/grub2/grub.cfg"
+        chk_bin_dir = "chk_bin"
+        chk_bin_folder = f"/tmp/{chk_bin_dir}"
+        chk_bin_src_path = f"/tmp/{chk_bin_dir}.tar.gz"
+
+        chk_loader_dir = "chk_loader"
+        chk_loader_folder = f"/tmp/{chk_loader_dir}"
+        chk_loader_src_path = f"/tmp/{chk_loader_dir}.tar.gz"
+
+        # Copy artifacts on to the node
+        node.shell.copy(PurePath(chkbinpath), PurePath(chk_bin_src_path))
+        node.shell.copy(PurePath(chkloaderpath), PurePath(chk_loader_src_path))
+        node.execute("sudo ls -lrt /tmp/chk_loader")
+        node.execute("sudo ls -lrt /tmp/chk_bin")
```

Comment:
Can we use Ls tool here, and specify run as sudo as parameter for method execute?

Reply:
This was temporary. I removed it.